src/platform/user/profile/vet360/containers/VAPProfileField.jsx

```diff
@@ -0,0 +1,366 @@
+      'profile-section': this.props.analyticsSectionName,
+    });
+  }
+
+  isEditLinkVisible = () => {
+    let transactionPending = false;
+    if (this.props.transaction) {
+      transactionPending = isPendingTransaction(this.props.transaction);
+    }
+    return (
+      !this.props.isEmpty &&
+      (!transactionPending ||
+        this.props.showEditView ||
+        this.props.showValidationView)
+    );
+  };
+
+  render() {
+    const {
+      analyticsSectionName,
+      fieldName,
+      showEditView,
+      isEmpty,
+      ContentView,
+      EditView,
+      ValidationView,
+      showValidationView,
+      title,
+      transaction,
+      transactionRequest,
+    } = this.props;
+
+    const childProps = {
+      ...this.props,
+      refreshTransaction: this.refreshTransaction,
+      clearErrors: this.clearErrors,
+      onAdd: this.onAdd,
+      onEdit: this.onEdit,
+      onChangeFormDataAndSchemas: this.onChangeFormDataAndSchemas,
+      onDelete: this.onDelete,
+      onCancel: this.onCancel,
+      onSubmit: this.onSubmit,
+    };
+
+    // default the content to the read-view
+    let content = (
+      <div className={classes.wrapper}>
+        <ContentView data={this.props.data} />
+        <VAPEditButton
+          onEditClick={this.isEditLinkVisible() ? this.onEdit : null}
+          fieldName={fieldName}
+          title={title}
+          className={classes.editButton}
+        />
+      </div>
+    );
+
+    if (isEmpty) {
+      content = (
+        <button
+          type="button"
+          onClick={this.onAdd}
+          className="va-button-link va-profile-btn"
+        >
+          Please add your {title.toLowerCase()}
+        </button>
+      );
+    }
+
+    if (showEditView) {
+      content = <EditView {...childProps} />;
+    }
+
+    if (showValidationView) {
+      content = (
+        <ValidationView
+          transaction={transaction}
+          transactionRequest={transactionRequest}
+          title={title}
+          clearErrors={this.clearErrors}
+        />
+      );
+    }
+
+    return (
+      <div className="vet360-profile-field" data-field-name={fieldName}>
+        <Vet360Transaction
+          isModalOpen={showEditView || showValidationView}
+          id={`${fieldName}-transaction-status`}
+          title={title}
+          transaction={transaction}
+          transactionRequest={transactionRequest}
+          refreshTransaction={() =>
+            this.props.refreshTransaction(transaction, analyticsSectionName)
+          }
+        >
+          {content}
+        </Vet360Transaction>
+      </div>
+    );
+  }
+}
+
+export const mapStateToProps = (state, ownProps) => {
+  const { fieldName } = ownProps;
+  const { transaction, transactionRequest } = selectVet360Transaction(
+    state,
+    fieldName,
+  );
+  const data = selectVet360Field(state, fieldName);
+  const isEmpty = !data;
+  const addressValidationType =
+    state.vet360.addressValidation.addressValidationType;
```

Comment:
sure. will do. also just made this ticket since I was working in the vet360 selectors file https://github.com/department-of-veterans-affairs/va.gov-team/issues/9548